repos/spack_repo/builtin/packages/cdi/package.py

```diff
@@ -24,6 +43,14 @@ class Cdi(AutotoolsPackage):
 
     depends_on("netcdf-c", when="+netcdf")
 
+    def url_for_version(self, version):
+        resource_id = RESOURCE_ID.get(str(version))
+        if resource_id is None:
+            raise InstallError(f"Download link mapping for {version} has not been added yet.")
```

Comment:
(nitpick) Minor change if you like it. It basically says "`cdi`" in the error message:
```suggestion
            raise InstallError(f"don't know which url to use to download cdi@{version}")
```